mitosheet/mitosheet/saved_analyses/upgrade.py

```diff
@@ -358,6 +358,18 @@ def upgrade_saved_analysis_to_have_up_to_date_args(saved_analysis: Optional[Dict
     
     return saved_analysis
 
+def upgrade_saved_analysis_to_have_code(saved_analysis: Optional[Dict[str, Any]], code: Optional[List[str]]) -> Optional[Dict[str, Any]]:
+
+    if saved_analysis is None:
+        return None
+    
+    if code is not None:
+        saved_analysis['code'] = code
+    else:
+        saved_analysis['code'] = []
```

Comment:
Feels like we need to handle this case in the code that gets form the backend as well. If the old code is empty, and there's code there, we shouldn't detect that they edited things (or write a new cell). Maybe we want to tag this empty case with 'empty' or something?